composer/trainer/trainer.py

```diff
@@ -1456,15 +1456,16 @@ def _get_autoresume_checkpoint(
             f'Looking for autoresume checkpoint: {save_latest_remote_file_name} (remote), {latest_checkpoint_path} (local)'
         )
 
-        if self.deepspeed_enabled:
+        # DeepSpeed and FSDP sharded state dict save checkpoints on every rank
+        if self.deepspeed_enabled or self.state.fsdp_sharded_state_dict_enabled:
```

Comment:
did you test autoresume + FSDP sharded? I think that should be unit testable...

Reply:
@eracah do u wanna add it :) I did not test it

Reply:
ok added unit test for sharded + autoresume and fixed a bug the unit test discovered 😉